src/core/router/index.js

```diff
@@ -26,23 +26,27 @@ const router = new Router({
 });
 router.beforeResolve((to, from, next) => {
   // Check if user is coming from a path that needs auth
-  if (
-    !from.meta.hasOwnProperty('requiresAuth') &&
-    to.meta.hasOwnProperty('requiresAuth')
-  ) {
+  if (store.state.wallet.address && to.meta.hasOwnProperty('requiresAuth')) {
     store.dispatch('wallet/removeWallet');
   }
   if (to.meta.hasOwnProperty('requiresAuth')) {
     next();
   } else {
     if (store.state.wallet.address === null) {
       store.dispatch('external/setLastPath', to.path);
-      next({ name: 'AccessWallet' });
+      /** only resolve a new path when current path isn't accesswallet
+       * this is a little hacky but utilizes intended purpose.
+       * vue router notes: move on to the next hook in the pipeline. If no hooks are left, the navigation is confirmed.
+       */
+      if (from.name !== 'AccessWallet') router.push({ name: 'AccessWallet' });
```

Comment:
But what if you're on access wallet and want to use the dropdown again?  At that point the routing wouldn't work.  Hmm...